crates/node-core/src/version.rs

```diff
@@ -11,6 +16,35 @@ pub const VERGEN_GIT_SHA: &str = env!("VERGEN_GIT_SHA");
 /// The build timestamp.
 pub const VERGEN_BUILD_TIMESTAMP: &str = env!("VERGEN_BUILD_TIMESTAMP");
 
+//The identification of the client
+/// - The two letter client code
+/// - The human readable name of the client
+/// - The version string of the current implementation 
+/// - first four bytes of the latest commit hash of this build
+/// 
+/// # Example
+/// 
+///  ```text
+/// code: Rh
+/// name: reth
+/// version: 0.1.0
+/// commit: defa64b2
+///  ```
+pub const CLIENTVERSIONV1: &str = const_str::concat(
```

Comment:
this should be a struct, the spec expects us to send a json object over the wire that looks like

```js
{
  code: "RH",
  name: "Reth",
  version: "v0.2.0-beta.5",
  commit: "defa64b2"
}
```

Reply:
hey , when i am trying to compile I am getting two errors `environment variable `NAME_CLIENT` not defined at compile time` and `environment variable `CLIENT_CODE` not defined at compile time` . I have also added a struct and using that as `CLIENTVERSIONV1` type.

Reply:
hey! that's because those two are not environment variables, they are constants. feel free to push your most recent changes with the struct and i'll take a look:)

Reply:
I have pushed the changes.

Reply:
hey @onbjerg , i tried debugging though those errors and after hardcoding some values the errors were solved except for a EngineApi instance in ` crates/node-builder/src/launch/mod.rs`. Can you give me some pointers on so that i can use `NAME_CLIENT ` and `CLIENT_CODE` as env in compile time.

Reply:
i'm not keen on having those as environment variables as it complicates building and testing

Reply:
Hey, so I already changed those and in the engine_api.rs I imported the values from the reth_node_core crate but when I try to add to cargo.toml it's showing me a cyclic dependency error (it works without error if I don't add it to cargo.toml). Also when I try the same thing in `crates/node-builder/src/launch/mod.rs` , its showing me `no `CLIENTVERSIONV1` in the root`.

Reply:
> i'm not keen on having those as environment variables as it complicates building and testing

Hey, so I already changed those and in the engine_api.rs I imported the values from the reth_node_core crate but when I try to add to cargo.toml it's showing me a cyclic dependency error (it works without error if I don't add it to cargo.toml). Also when I try the same thing in `crates/node-builder/src/launch/mod.rs `, its showing me no CLIENTVERSIONV1 in the root.

Reply:
sure, open a new pr from a diff branch, this will also allow me to push to it

let's pick up the discussion in the new pr